### ISSUE TITLE
Should `prefer-namespace-import` be triggered on type?

### ISSUE THREAD

**Issue by Sparticuz** [marcalexiei/eslint-plugin-zod-x#38]
### 🧩 Description

<img width="866" height="75" alt="Image" src="https://github.com/user-attachments/assets/a9a6b1ba-ad76-480d-937b-55c13342ab7a" />

### ⚙️ Options

n/a

### 🧾 Example Configuration

Recommended config

### ✅ Valid Code Examples

```typescript
n/a
```

### ❌ Invalid Code Examples

```typescript
n/a
```
